src/Toolkit/PubSub/V1/PublisherApi.php

```diff
@@ -0,0 +1,673 @@
+     *     $response = $publisherApi->getTopic($formattedTopic);
+     * } finally {
+     *     if (isset($publisherApi)) {
+     *         $publisherApi->close();
+     *     }
+     * }
+     * </code></pre>
+     *
+     * @param string $topic        The name of the topic to get.
+     * @param array  $optionalArgs {
+     *                             Optional. There are no optional parameters for this method yet;
+     *                             this $optionalArgs parameter reserves a spot for future ones.
+     *                             }
+     * @param array  $callSettings {
+     *                             Optional.
+     *
+     *    @var Google\GAX\RetrySettings $retrySettings
+     *          Retry settings to use for this call. If present, then
+     *          $timeout is ignored.
+     *    @var int $timeoutMillis
+     *          Timeout to use for this call. Only used if $retrySettings
+     *          is not set.
+     * }
+     *
+     * @return google\pubsub\v1\Topic
+     *
+     * @throws Google\GAX\ApiException if the remote call fails
+     */
+    public function getTopic($topic, $optionalArgs = [], $callSettings = [])
+    {
+        $request = new GetTopicRequest();
+        $request->setTopic($topic);
+
+        $mergedSettings = $this->defaultCallSettings['getTopic']->merge(
+            new CallSettings($callSettings)
+        );
+        $callable = ApiCallable::createApiCall(
+            $this->stub,
+            'GetTopic',
+            $mergedSettings,
+            $this->descriptors['getTopic']
+        );
+
+        return $callable(
+            $request,
+            [],
+            ['call_credentials_callback' => $this->createCredentialsCallback()]);
+    }
+
+    /**
+     * Lists matching topics.
+     *
+     * Sample code:
+     * <pre><code>
+     * try {
+     *     $publisherApi = new PublisherApi();
+     *     $formattedProject = PublisherApi::formatProjectName("[PROJECT]");
+     *     for ($publisherApi->listTopics($formattedProject) as element) {
+     *         // doThingsWith(element);
+     *     }
+     * } finally {
+     *     if (isset($publisherApi)) {
+     *         $publisherApi->close();
+     *     }
+     * }
+     * </code></pre>
+     *
+     * @param string $project      The name of the cloud project that topics belong to.
+     * @param array  $optionalArgs {
+     *                             Optional.
+     *
+     *     @var int $pageSize Maximum number of topics to return.
+     * }
+     *
+     * @param array $callSettings {
+     *                            Optional.
+     *
+     *    @var Google\GAX\RetrySettings $retrySettings
+     *          Retry settings to use for this call. If present, then
+     *          $timeout is ignored.
+     *    @var int $timeoutMillis
+     *          Timeout to use for this call. Only used if $retrySettings
+     *          is not set.
+     * }
+     *
+     * @return Google\GAX\PageAccessor
```

Comment:
Would it be possible to have `list` calls that also need to be paginated manually instead of always returning an iterator?